lib/src/main/java/com/orange/ods/compose/component/utilities/Preview.kt

```diff
@@ -44,3 +46,49 @@ internal open class BasicPreviewParameterProvider<T>(vararg values: T) : Preview
  * @param clazz The enum class.
  */
 internal open class EnumPreviewParameterProvider(clazz: KClass<out Enum<*>>) : BasicPreviewParameterProvider<Enum<*>>(*clazz.java.enumConstants)
+
+/**
+ * Multipreview annotation classes used to display both light and dark mode previews.
+ *
+ * The only reason why `UiModePreviews` is an annotation class is to colorize it as an annotation in Android Studio.
+ * An empty `Target` annotation has been added in order to avoid using the parent `UiModePreviews` annotation which has no effect.
+ */
+@Target
+internal annotation class UiModePreviews {
+
+    companion object {
+        private const val LightName = "Light"
+        private const val DarkName = "Dark"
+        private const val ButtonWidthDp = 200
+        private const val ChipWidthDp = 180
+        private const val TabWidthDp = 100
+    }
+
+    @Preview(name = LightName)
+    @Preview(name = DarkName, uiMode = Configuration.UI_MODE_NIGHT_YES, showBackground = true)
+    annotation class Default
+
+    @Preview(name = LightName, widthDp = ButtonWidthDp)
+    @Preview(name = DarkName, uiMode = Configuration.UI_MODE_NIGHT_YES, showBackground = true, widthDp = ButtonWidthDp)
+    annotation class OdsButton
+
+    @OdsButton
+    annotation class OdsOutlinedButton
+
+    @OdsButton
+    annotation class OdsTextButton
```

Comment:
You created a preview by component, do we have to add a preview each time we add a component? 
I was wondering if this would be better if we had only 4 annotation classes : `Default`, `Button`, `Chip` and `Tab`
What do you think?

Reply:
That's what I did first. But then I realized that `OdsIconToggleButton` preview does no use a specific width. That's why I created a specific annotation for each component. I must say I don't know which solution is better.

Reply:
I understand but this can be painful to add a new annotation class each time which is only a reference to another one. Is it a problem to keep the `Default` for `OdsIconToggleButton`? Or we can add a specific annotation for this one called `IconToggleButton` referencing the `Default` one?